backend/src/models/userCreds.js

```diff
@@ -9,8 +10,49 @@ const userCredsSchema = mongoose.Schema(
     refreshToken: { type: String },
     userDataId: { type: ObjectId }
   },
-  { timestamps: true }
+  {
+    timestamps: true
+    // toJSON: {
+    //   getters: true,
+    //   virtuals: true
+    // }
+  }
 );
 
+userCredsSchema.statics.upsertGoogleUser = function(accessToken, refreshToken, profile, cb) {
+  const that = this;
+  return this.findOne(
+    {
+      googleId: profile.id
+    },
+    (err, user) => {
+      // no user was found, lets create a new one
+      if (!user) {
+        const { email, given_name, family_name } = profile._json;
+        const newUserData = new UserData({
+          role: 'new',
+          bio: { email, first_name: given_name, last_name: family_name }
+        });
+
+        newUserData.save((err1, savedUserData) => {
+          const newUser = new that({
+            googleId: profile.id,
+            accessToken,
+            refreshToken,
+            userDataId: savedUserData.id
+          });
+          newUser.save((error, savedUser) => {
+            if (error) console.log(error);
+            return cb(error, savedUser);
+          });
+        });
+      } else {
+        // user found.
+        return cb(err, user);
```

Comment:
We're not updating anything if the user already exists in the system. Is this intended? If so, I don't think `upsert` is a good name for the function.

Reply:
I see your point. I don't think think we need to update anything if the user already exists? 

Maybe I'm missing something... but I think we just want to:
1) create a UserCreds & UserData object if they don't exist,
2) or return the UserCreds if it does exist

Do you have a suggestion for what it should be called?

Reply:
I guess it could be called `UserCreds.findOrCreate(...)`? Gonna change it to that for now, if you have another suggestion I'm open to change it again!

Reply:
That sounds like a better name!